src/radical/pilot/unit_manager.py

```diff
@@ -996,14 +996,19 @@ def wait_units(self, uids=None, state=None, timeout=None):
 
         self._rep.progress_done()
 
-        if to_check: self._rep.warn('>>timeout\n')
-        else       : self._rep.ok  ('>>ok\n')
 
         # grab the current states to return
         state = None
         with self._units_lock:
             states = [self._units[uid].state for uid in uids]
 
+        sdict = {state: states.count(state) for state in set(states)}
+        for state in sorted(set(states)):
+            self._rep.info('\t%-10s: %5d\n' % (state, sdict[state]))
+
+        if to_check: self._rep.warn('>>timeout\n')
+        else       : self._rep.ok  ('>>ok\n')
+
```

Comment:
Oh, another unrelated change which would have warranted it's own PR - sorry!  It triggers the reported to print a state summary after `wait_units()`.